src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.VirtualMonoRepo.E2E.Tests/VmrTestsBase.cs

```diff
@@ -0,0 +1,272 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using CSharpx;
+using FluentAssertions;
+using Microsoft.DotNet.DarcLib.Helpers;
+using NUnit.Framework;
+
+#nullable enable
+namespace Microsoft.DotNet.Darc.Tests.VirtualMonoRepo;
+
+public abstract class VmrTestsBase : GitOperations
+{
+    protected LocalPath _currentTestDirectory = null!;
+    protected LocalPath _privateRepoPath = null!;
+    protected LocalPath _vmrPath = null!;
+    protected LocalPath _tmpPath = null!;
+    protected LocalPath _externalRepoPath = null!;
+    protected LocalPath _dependencyRepoPath = null!;
+    protected LocalPath _specialRepoPath = null!;
+    protected LocalPath _installerRepoPath = null!;
+
+
+    [SetUp]
+    public async Task Setup()
+    {
+        _currentTestDirectory = VmrTestsOneTimeSetUp.TestsDirectory / Path.GetRandomFileName();
+        Directory.CreateDirectory(_currentTestDirectory);
+
+        _tmpPath = _currentTestDirectory / "tmp";
+        _privateRepoPath = _currentTestDirectory / "test-repo";
+        _vmrPath = _currentTestDirectory / "vmr";
+        _externalRepoPath = _currentTestDirectory / "external-repo";
+        _dependencyRepoPath = _currentTestDirectory / "dependency";
+        _specialRepoPath = _currentTestDirectory / "special-repo";
+        _installerRepoPath = _currentTestDirectory / "installer";
+        _externalRepoPath = _currentTestDirectory / "external-repo";
+
+        Directory.CreateDirectory(_tmpPath);
+
+        await CopyVmrForCurrentTest();
+        await CopyReposForCurrentTest();
+    }
+
+    [TearDown]
+    public void DeleteCurrentTestDirectory()
+    {
+        try
+        {
+            if (_currentTestDirectory is not null)
+            {
+                DeleteDirectory(_currentTestDirectory.ToString());
+            }
+        }
+        catch
+        {
+            // Ignore
+        }
+    }
+
+    protected abstract Task CopyReposForCurrentTest();
+
+    protected abstract Task CopyVmrForCurrentTest();
+
+    protected List<LocalPath> GetExpectedFilesInVmr(
+        LocalPath vmrPath,
+        string[] syncedRepos,
+        List<LocalPath> reposFiles)
+    {
+        var expectedFiles = new List<LocalPath>
+        {
+            vmrPath / "git-info" / "AllRepoVersions.props",
+            vmrPath / "src" / "source-manifest.json",
+            vmrPath / "src" / "source-mappings.json",
+        };
+
+        foreach (var repo in syncedRepos)
+        {
+            expectedFiles.Add(vmrPath / "src" / repo / "eng" / Constants.VersionDetailsName);
+            expectedFiles.Add(vmrPath / "git-info" / $"{repo}.props");
```

Comment:
All `git-info` names should come from `VmrInfo` where we have the name of this folder (it will actually change soon so that's why we care)